test_runner/regress/test_subscriber_branching.py

```diff
@@ -239,3 +239,148 @@ def insert_data(pub, start):
             res = scur_postgres.fetchall()
             assert len(res) == 1
             assert str(sub_child_2_timeline_id) == res[0][0]
+
+
+def test_multiple_subscription_branching(neon_simple_env: NeonEnv):
```

Comment:
Just a question: does this test repro issue more or less robustly? Or it's still very probabilistic and several runs are needed to repro it?

Reply:
for me it reproduced the deadlock reliably every time